src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,53 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.InvalidInputException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int REQUIRED_AGE = 18;
+    private static final int MINIMUM_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
+    private final User user = new User();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new InvalidInputException("No data provided");
+        }
+        loginValidation();
+        passwordValidation();
+        ageValidation();
+        return storageDao.add(user);
+    }
+
+    private void loginValidation() {
+        if (user.getLogin() == null) {
+            throw new InvalidInputException("Please, fill all the required fields");
+        }
+        if (user.getLogin().length() < MINIMUM_LENGTH) {
+            throw new InvalidInputException("Your login should be at least 6 characters long");
```

Comment:
```suggestion
            throw new InvalidInputException("Your login should be at least " + MINIMUM_LENGTH + " characters long");
```

Reply:
This will result in creating more string objects each concatination

Reply:
agree. better to use a constant for the message, if you change the constant value you need to change the message too (check it in other messages)